src/autogluon/bench/runbenchmark.py

```diff
@@ -205,42 +213,47 @@ def wait_for_jobs(
     quit_statuses: Optional[list] = ["SUCCEEDED", "FAILED"],
     frequency: Optional[int] = 120,
 ):
+    if config_file is not None:
+        with open(config_file, "r") as f:
+            config = yaml.safe_load(f)
+            job_ids = list(config.get("job_configs", {}).keys())
+            aws_region = config.get("CDK_DEPLOY_REGION", aws_region)
+
+    batch_client = boto3.client("batch", region_name=aws_region)
     while True:
         all_jobs_completed = True
-        failed_jobs = []
+        failed_jobs = set()
 
         try:
-            job_status = get_job_status(job_ids=job_ids, cdk_deploy_region=aws_region, config_file=config_file)
-
-            for job_id, job_status in job_status.items():
-                if job_status == "FAILED":
-                    failed_jobs.append(job_id)
-                elif job_status not in quit_statuses:
-                    all_jobs_completed = False
+            job_status = get_job_status(job_ids=job_ids, cdk_deploy_region=aws_region, config_file=None)
+
+            for job_id, status in job_status.items():
+                if isinstance(status, str):
+                    if status == "FAILED":
+                        failed_jobs.append(job_id)
+                    elif status not in quit_statuses:
+                        all_jobs_completed = False
+                elif isinstance(status, dict):
+                    for status, num in status.items():
+                        if status == "FAILED" and num > 0:
+                            paginator = batch_client.get_paginator("list_jobs")
+                            for page in paginator.paginate(arrayJobId=job_id, jobStatus="FAILED"):
+                                for job in page["jobSummaryList"]:
+                                    failed_jobs.add(job["jobId"])
+                        if status not in quit_statuses and num > 0:
+                            all_jobs_completed = False
         except botocore.exceptions.ClientError as e:
             logger.error(f"An error occurred: {e}.")
             return
 
         if all_jobs_completed:
             break
         else:
-            time.sleep(frequency)  # Poll job statuses every 60 seconds
+            time.sleep(frequency)  # Poll job statuses every 120 seconds
```

Comment:
why do we increase the poll time?

Reply:
It's been 120s for a while but I forgot to update the comment.